ibis/backends/tests/test_client.py

```diff
@@ -1360,6 +1360,29 @@ def test_overwrite(ddl_con, monkeypatch):
         assert t2.count().execute() == expected_count
 
 
+@contextlib.contextmanager
+def create_and_destroy_db(con):
+    con.create_database(dbname := gen_name("db"))
+    try:
+        yield dbname
+    finally:
+        con.drop_database(dbname)
+
+
+# TODO: move this to something like `test_ddl.py`
+def test_insert_with_database_specified(con_create_database, monkeypatch):
+    con = con_create_database
+    monkeypatch.setattr(ibis.options, "default_backend", con)
```

Comment:
Why do we need to patch this?

Reply:
`t` is never executed, only compiled, and the compilation will use the dialect of `con`.